File: 基础语法补充/3-OC调用Swift代码/OC调用Swift代码/ViewController.m
//
//  ViewController.m
//  OC调用Swift代码
//
//  Created by 韩俊强 on 2017/7/13.
//  Copyright © 2017年 HaRi. All rights reserved.
//

#import "ViewController.h"
#import "OC调用Swift代码-Swift.h" // 引入Swift头文件 #import "工程名-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    //1.在OC项目中创建一个swift文件的时候，Xcode 会提示 需要创建一个桥接文件，点确定创建桥接文件，Xcode会自动创建一个桥接文件, 其实OC调用Swift这个桥接文件作用并不大, 可以不创建;
    
    //2.进入TARGETS ->Build Settings -> Packaging 中 设置Defines Module为YES; 如图:readMe.png
    
    //3.设置 Product Module Name ，也可以不设置，默认为工程的名字。这个在后面会用到
    //4.在swift写一个类,注意这个类一定要继承NSObject，不然在OC中没法用
    
    //5.在OC需要用到的swift文件中 导入文件 "Product Module Name -Swift.h"  因为 Product Module Name默认是工程的名字, 所以直接导入 #import "工程名-Swift.h"
    
    Person *p = [[Person alloc]init];
    p.name = @"小韩哥";
    NSLog(@"%@",p.name);
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 基础语法补充/Swift调用OC代码/Swift调用OC代码/ViewController.swift
//
//  ViewController.swift
//  Swift调用OC代码
//
//  Created by 韩俊强 on 2017/7/13.
//  Copyright © 2017年 HaRi. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        //1.当你在Swift项目中创建OC类或者别的OC代码时, 会自动生成一个桥接 项目名称-Bridging-Header.h 文件, 如图: readMe.png ;
        
        //2.将需要调用的OC头文件写入桥接文件, 强打头文件注意书写正确, 编译一下, 即可调用;
        
        //3.调用示例如下:
        
        let p = Person()
        p.name = "小韩哥"
        
        let phone = Phone()
        phone.color = UIColor.blue
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

